drivers/console/uart_mux.c

```diff
@@ -229,6 +229,10 @@ static int uart_mux_init(const struct device *dev)
 {
 	struct uart_mux_dev_data *dev_data = dev->data;
 
+	if (dev == NULL || dev_data == NULL) {
```

Comment:
1. The dev being NULL will already have caused problems at this point, so this check is pointless
2. The device driver is defined with config/data, so when should they be `NULL`?

Reply:
Not sure tbh, there are checks for some or all of `dev`, `dev_data`, `dev_data->dev`, so none of them are actually required?